.github/workflows/python-pull-request.yml

```diff
@@ -35,17 +38,16 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
-        cache: 'pip'
 
-    - name: Install test dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install pytest pytest-md pytest-emoji
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
+      with:
+        version: 0.7.19
 
     - name: Install packages
       run: |
         # Install all packages from local
-        pip install -e ./training[all,tests] -e ./graphs[all,tests] -e ./models[all,tests]
+        uv pip install -e ./training[all,tests] -e ./graphs[all,tests] -e ./models[all,tests] pytest pytest-md pytest-emoji
```

Comment:
what's `pytest pytest-md pytest-emoji` doing?

Reply:
These are used by the pytest action. The pytest is a bit redundant but it doesn't hurt to have it just in case,  pytest-md generates test reports and pytest-emoji to print emojis in the log depending on pass/fail 😀

Example:
<img width="500" alt="image" src="https://github.com/user-attachments/assets/71496763-7518-4ff2-8beb-f498a53b8054" />